src/main/java/io/vertx/core/eventbus/impl/clustered/ClusteredMessage.java

```diff
@@ -117,15 +116,14 @@ public Buffer encodeToWire() {
       // User codec
       writeString(buffer, messageCodec.name());
     }
-    buffer.appendByte(send ? (byte)0 : (byte)1);
+    buffer.appendByte(send ? (byte) 0 : (byte) 1);
     writeString(buffer, address);
     if (replyAddress != null) {
       writeString(buffer, replyAddress);
     } else {
       buffer.appendInt(0);
     }
-    buffer.appendInt(sender.port);
-    writeString(buffer, sender.host);
+    writeString(buffer, sender);
```

Comment:
this seems to mean that the wire protocol has changed between V3 and V4, is that right ?

Reply:
Yes

Reply:
I think we should revert this change and keep the protocol stable, it does not seem to be a valid enough reason to break it.

Reply:
if we really have to break it then we should discuss it

Reply:
The reason it changed is that nodes are no longer identified in the clustered eventbus by host/port. This solves a few issues (for example when a crashed nodes is restarted before all nodes in the cluster could notice).

Anyway compatibility is already broken because:

- the storage implementation is going to change in cluster managers
- we are going to upgrade HZ, ISPN and Ignite and they are not necessarily compatible with previous versions

Reply:
ok, then we should proceed in a way that

1/ we version the protocol
2/ we have a way that a V4 clustering node detects a V3 node and take an appropriate measure to handle it (logging, close connection, etc...) and vice versa

Reply:
Could you please file an issue for this that we'd address in a later milestone? It doesn't sound like a blocker for this one to be merged and cluster managers to be updated.

Reply:
yes it is not blocker, we need to find a way to implement this

Reply:
https://github.com/eclipse-vertx/vert.x/issues/3406